spec/services/pdf_parser_spec.rb

```diff
@@ -0,0 +1,19 @@
+require 'spec_helper'
+require_relative '../../app/services/pdf_parser'
+require './app/models/user'
+
+include AuthenticationSpecHelpers
+
+RSpec.describe PdfParser do
+  context 'when the PDF file is valid' do
+    let(:pdf_file) { Rack::Test::UploadedFile.new('spec/fixtures/sample.pdf', 'application/pdf') }
+    let!(:user) { authenticated }
```

Comment:
We don't need them to be authenticated for these tests, these should just be focused on the PdfParser. As long as the User is in the DB (via FactoryBot.create(:user) then it should work.